components/core/src/clp/utils.cpp

```diff
@@ -201,31 +193,32 @@ namespace clp {
     }
 
     void split_file (const string& path_for_compression, group_id_t group_id, const TimestampPattern* last_timestamp_pattern,
-                     streaming_archive::writer::Archive& archive_writer, streaming_archive::writer::File*& file)
+                     streaming_archive::writer::Archive& archive_writer)
     {
-        auto orig_file_id = file->get_orig_file_id();
-        auto split_ix = file->get_split_ix();
-        file->set_is_split(true);
-        close_file_and_mark_ready_for_segment(archive_writer, file);
+        const auto& encoded_file = archive_writer.get_file();
+        auto orig_file_id = encoded_file.get_orig_file_id();
+        auto split_ix = encoded_file.get_split_ix();
+        archive_writer.set_file_is_split(true);
+        close_file_and_append_to_segment(archive_writer);
 
-        file = create_and_open_file(archive_writer, path_for_compression, group_id, orig_file_id, ++split_ix);
+        archive_writer.create_and_open_file(path_for_compression, group_id, orig_file_id, ++split_ix);
         // Initialize the file's timestamp pattern to the previous split's pattern
-        archive_writer.change_ts_pattern(*file, last_timestamp_pattern);
+        archive_writer.change_ts_pattern(last_timestamp_pattern);
     }
 
     void split_file_and_archive (streaming_archive::writer::Archive::UserConfig& archive_user_config, const string& path_for_compression, group_id_t group_id,
-                                 const TimestampPattern* last_timestamp_pattern, streaming_archive::writer::Archive& archive_writer,
-                                 streaming_archive::writer::File*& file)
+                                 const TimestampPattern* last_timestamp_pattern, streaming_archive::writer::Archive& archive_writer)
     {
-        auto orig_file_id = file->get_orig_file_id();
-        auto split_ix = file->get_split_ix();
-        file->set_is_split(true);
-        close_file_and_mark_ready_for_segment(archive_writer, file);
+        const auto& encoded_file = archive_writer.get_file();
+        auto orig_file_id = encoded_file.get_orig_file_id();
```

Comment:
`const auto&` to avoid a copy

Reply:
I believe for this one we can't use reference because the file will be freed when closing the archive.

Reply:
Oh right, sorry.